test/Base_Tests/src/Data/Numbers_Spec.enso

```diff
@@ -324,21 +348,21 @@ add_specs suite_builder =
             (1 < 0.99).should_be_false
             (3 < hundred_factorial).should_be_true
             (3 < very_negative).should_be_false
-            (3 < Nothing).should_fail_with Incomparable_Values
+            Test.expect_panic_with (3 < Nothing) Type_Error
```

Comment:
Why does this become a `Type_Error`?

Reply:
I think this is problematic from the perspective of `Vector.sort`. Does this not change it's semantics when sorting `["a", 2].sort`? or it's handled separately and unaffected?

Reply:
If sort uses `Ordering.compare` (I believe it does), then it should be unaffected.

Reply:
Alas, the `Type_Error` comes from the engine infrastructure and cannot be easily replaced by a different error. Also there is nobody between the caller of `<` and the infrastructure to catch the exception and turn it to something else.

I had to modify following tests: https://github.com/enso-org/enso/pull/8779/commits/46b7f8172239bdd3ac0c1e2989a5747172bfe519